[BUG] Unexpected update of a btp_subaccount_service_binding resource.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of the Terraform provider are you using?

1.7.0

### What version of the Terraform CLI are you using?

OpenTofu v1.8.1

### What type of issue are you facing

bug report

### Describe the bug

I'm implementing a service binding rotation.
```
resource "btp_subaccount_service_binding" "service_manager_service_binding" {
  provider            = btp
  subaccount_id       = btp_subaccount.subscription_test.id
  service_instance_id = btp_subaccount_service_instance.service_manager.id
  name                = local.service_manager_binding_name
  depends_on          = [btp_subaccount_service_instance.service_manager]
  labels = {
    rotationVersion = [local.expected_credentials_rotation_version]
  }
  lifecycle {
    replace_triggered_by = [null_resource.trigger_rotation]
  }
}
```
Because local.expected_credentials_rotation_version is changed, the provider tries to update the service binding and fails with the following error.
The instance recreation would be triggered by lifecycle/replace_triggered_by, so there is no need for update of old service binding. 

```bash
24-10-29T09:24:29.700Z] module.cloudci.module.subscription-test-subaccount[0].btp_subaccount_service_binding.service_manager_service_binding: Modifying... [id=f5b880c4-2800-4fd6-933e-39e0388e1757]

[2024-10-29T09:24:29.700Z] 2024-10-29T09:24:29.594Z [ERROR] provider.terraform-provider-btp_v1.7.0: Response contains error diagnostic: @module=sdk.proto diagnostic_detail="This resource is not supposed to be updated" diagnostic_severity=ERROR tf_provider_addr=registry.terraform.io/sap/btp @caller=github.com/hashicorp/terraform-plugin-go@v0.24.0/tfprotov6/internal/diag/diagnostics.go:58 diagnostic_summary="API Error Updating Resource Service Binding (Subaccount)" tf_proto_version=6.6 tf_req_id=a7db2a09-8e83-ca0d-d019-fcefac3d49be tf_resource_type=btp_subaccount_service_binding tf_rpc=ApplyResourceChange timestamp=2024-10-29T09:24:29.594Z
[2024-10-29T09:24:29.700Z] 2024-10-29T09:24:29.596Z [ERROR] vertex "module.cloudci.module.subscription-test-subaccount[0].btp_subaccount_service_binding.service_manager_service_binding" error: API Error Updating Resource Service Binding (Subaccount)
[2024-10-29T09:24:30.273Z] ╷
[2024-10-29T09:24:30.273Z] │ Error: API Error Updating Resource Service Binding (Subaccount)
[2024-10-29T09:24:30.273Z] │ 
[2024-10-29T09:24:30.273Z] │   with module.cloudci.module.subscription-test-subaccount[0].btp_subaccount_service_binding.service_manager_service_binding,
[2024-10-29T09:24:30.273Z] │   on .terraform/modules/cloudci/modules/subscription-test-subaccount/main.tf line 176, in resource "btp_subaccount_service_binding" "service_manager_service_binding":
[2024-10-29T09:24:30.273Z] │  176: resource "btp_subaccount_service_binding" "service_manager_service_binding" {
[2024-10-29T09:24:30.273Z] │ 
[2024-10-29T09:24:30.273Z] │ This resource is not supposed to be updated
```


### Expected Behavior

I expect the provider not try to update some properties of binding if it is not supported. It should just try to recreate the binging in cases some property can not be updated.
Cloud foundry service keys resources behave in such way.

### Steps To Reproduce

1. Execute apply creating a service binding resource
2. Change name or add a label of the resource in configuration
3. Execute apply again

### User's Role Collections

_No response_

### Add screenshots to help explain your problem

_No response_

### Additional context

_No response_

--- a/internal/provider/resource_subaccount_service_binding.go
+++ b/internal/provider/resource_subaccount_service_binding.go
@@ -11,6 +11,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/path"
 	"github.com/hashicorp/terraform-plugin-framework/resource"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/mapplanmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
@@ -86,6 +87,10 @@ You must be assigned to the admin or the service administrator role of the subac
 				MarkdownDescription: "The set of words or phrases assigned to the service binding.",
 				Computed:            true,
 				Optional:            true,
+				PlanModifiers: []planmodifier.Map{
+					mapplanmodifier.RequiresReplace(),
+					mapplanmodifier.UseStateForUnknown(),
+				},
 			},
 			"id": schema.StringAttribute{
 				MarkdownDescription: "The ID of the service binding.",

--- a/internal/provider/resource_subaccount_service_binding.go
+++ b/internal/provider/resource_subaccount_service_binding.go
@@ -89,7 +89,6 @@ You must be assigned to the admin or the service administrator role of the subac
 				Optional:            true,
 				PlanModifiers: []planmodifier.Map{
 					mapplanmodifier.RequiresReplace(),
-					mapplanmodifier.UseStateForUnknown(),
 				},
 			},
 			"id": schema.StringAttribute{

